integration-tests/jcache/src/main/java/org/apache/camel/quarkus/component/jcache/it/JcacheCaffeineReflectionConfiguration.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.jcache.it;
+
+import io.quarkus.runtime.annotations.RegisterForReflection;
+
+@RegisterForReflection(classNames = {
```

Comment:
```suggestion
/** These reflection registrations should be moved to a new camel-quarkus-jcache-caffeine extension - see https://github.com/apache/camel-quarkus/issues/4092 */
@RegisterForReflection(classNames = {
```

Reply:
Actually I think the new `caffeine-jcache` extension should be part of the Quarkus itself, not `camel-quarkus`, because caffeine jcache has really nothing to do with Camel. I'll create another PR to Quarkus and if it is accepted, I'll remove this class altogether.

Reply:
There is value in merging this PR now and refactoring afterward. As you want.

Reply:
Sure, what I meant is if and after the Quarkus supports `caffeine-jcache`, we can revisit this one and remove this class. But for now I guess it's not big harm to have it here.

Reply:
> Actually I think the new `caffeine-jcache` extension should be part of the Quarkus itself, not `camel-quarkus`, because caffeine jcache has really nothing to do with Camel. I'll create another PR to Quarkus and if it is accepted, I'll remove this class altogether.

Good luck with it! I bet they'll recommend creating a new Quarkiverse repo for that, which I'd personally not prefer because of the maintenance overhead. Note that you'd also have to move the service provider registration to a new jcache extension.

Reply:
Hmm, yeah, I thought about this, but maybe the easiest would be just to add jcache support to the existing quarkus caffeine extension.
As for the service provider registration, no, I think it has to stay here, again, because there can be any provider, not just caffeine, it has to be generic, so should be discoverable by this extension.